internal/target.go

```diff
@@ -46,6 +46,14 @@ func (t Target) Scan() (sbom.SBOM, error) {
 		},
 	}
 
+	// Enable all the image catalogers to mimic same cataloging behavior as syft running on an image
+	config := cataloger.DefaultConfig()
+	catalogers := cataloger.ImageCatalogers(config)
```

Comment:
So because we also support scanning directories in the form of the build context using `BUILDKIT_SBOM_SCAN_CONTEXT` (see https://docs.docker.com/build/attestations/sbom/#arguments), I wonder if we'd stop detecting things from there that we might have otherwise found.

Would `AllCatalogers` present problems? Curious if maybe the same packages might be duplicated or something like that?

Alternatively, we could make changes to try and indicate to the scanner which of the directories is the context, so we could apply a different scanner? Though this requires some more significant changes to buildkit.

Reply:
I can run some experiments and see what the results are with `AllCatalogers`

Reply:
That would be super helpful :+1: We have some examples in the `examples/` directory, that might be interesting to you.

Reply:
ping @cdupuis did you manage to take a look at this?

Reply:
As a concrete example of where `ImageCataloguers` causes a regression: if I have a local context that contains `go.mod`, `ImageCatalogers` won't detect any go packages, while `DirectoryCataloguers` and `AllCataloguers` do. IMO we _probably_ want to be able to detect this case?